resources/prosody-plugins/mod_breakout_rooms.lua

```diff
@@ -0,0 +1,384 @@
+local jid_bare = require 'util.jid'.bare;
+local jid_node = require 'util.jid'.node;
+local jid_resource = require 'util.jid'.resource;
+local jid_split = require 'util.jid'.split;
+local json = require 'util.json';
+local st = require 'util.stanza';
+local uuid_gen = require 'util.uuid'.generate;
+
+local get_room_from_jid = module:require "util".get_room_from_jid;
+
+local BROADCAST_ROOMS_INTERVAL = .3;
```

Comment:
I will add some note. This interval just means that at most one message per 300 ms is sent. A message is only sent if there was a change of presence or the list of rooms. This is to avoid sending an update on every change when for example all participants are auto-assigned to breakout rooms or a room gets closed and this flooding the clients.